src/lib/snark_worker/prod.ml

```diff
@@ -130,6 +130,31 @@ module Impl = struct
           ~metadata ;
         Error e
 
+  let cache_and_time_partitioned ~logger ~cache ~statement
+      ~(partitioned_spec : Work.Spec.Partitioned.Stable.Latest.t) k =
+    match (Cache.find cache) statement with
+    | Some proof ->
+        Deferred.Or_error.return (proof, Time.Span.zero)
+    | None -> (
+        let start = Time.now () in
+        match%map.Async.Deferred
+          Monitor.try_with_join_or_error ~here:[%here] k
+        with
+        | Error e ->
+            [%log error] "SNARK worker failed: $error"
+              ~metadata:
+                [ ("error", Error_json.error_to_yojson e)
+                ; ( "partitioned_spec"
+                  , Work.Spec.Partitioned.Stable.Latest.to_yojson
+                      partitioned_spec )
+                ] ;
+            Error e
+        | Ok res ->
+            Cache.add cache ~statement ~proof:res ;
+            let elapsed = Time.abs_diff (Time.now ()) start in
+            Ok (res, elapsed) )
+
+  (* TODO: remove below after Snark Worker optimization has been fully merged *)
```

Comment:
You're using this function in `perform_partitioned`, but there is a comment to remove it.
I don't think we need a cache mechanism here, but let's then not use this function in the new code?